### PR TITLE
add ware unpack command

### PR DESCRIPTION
**Pull request by TripleDogDare**
Currently unpacks directly from workspace stack to the local filesystem and passes along arguments to rio to do the actual work.

### CODE REVIEW

**Comment by warpfork** [comment]
Definitely want for tests.

I think this could be done fairly nicely with a testmark exec situation which runs the unpack command, then simply procedes to use shell find and cat commands to visualize and sanity check the results.  Would also make good documentation examples!

**Comment by TripleDogDare** [comment]
:warning: Required before merging: A `--force` flag and checks that we don't explode an existing directory.

**Comment by TripleDogDare** [comment]
@warpfork New commit. Fixes all the CLI tests using my local branch of testexec. Which is significantly less frustrating, for what it's worth regarding the state of that PR.

**Comment by warpfork** [comment]
:eyes:  Looks nice.

I can easily see from the test fixture that this does sensible things as a CLI.  :+1: 

What happens if the `--path` argument isn't provided?

I wish these tests still worked offline.  I think it should be possible.  We vendored a (very short) list of assets in the git repo for that purpose.  Based on some of the diff and the context of other stuff I know you've been working on, though, I guess maybe some of this got decorated with "/net/" because that was the path of least resistance as you fixed some testrot in things that weren't getting exercised?

**Comment by TripleDogDare** [comment]
Without path it will unpack into the PWD assuming that PWD is an empty directory.

There's still some test rot that needs some fixing. The unpack tests aren't actually running.
I did push some more stuff under `/net/` as path of least resistance. I'll look into what we can do without that when the tests actually run.

**Comment by TripleDogDare** [comment]
This is now dependent on #39

**Review by warpfork — COMMENTED** [review]
Shoot, I did a partial review of this a while ago and then didn't click submit.  I think I have some comments that are already outdated, but I don't see a way in the github UI to remove _pending_ comments that are already outdated, so, uh, heh, sorry for those in advance.

File: cmd/warpforge/ware.go [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+
+var wareCmdDef = cli.Command{
+	Name:  "ware",
+	Usage: "Subcommands that operate on wares",
+	Subcommands: []*cli.Command{
+		{
+			Name:  "unpack",
+			Usage: "Places the contents of a ware onto the filesystem",
+			Description: strings.Join([]string{
+				`[WareID]: a ware ID such as [packtype]:[hash]. e.g. "tar:4z9DCTxoKkStqXQRwtf9nimpfQQ36dbndDsAPCQgECfbXt3edanUrsVKCjE9TkX2v9"`,
```
[the remark this answers is not in the document]
Comment (TripleDogDare) [reply cmd/warpforge/ware.go:30]: Yeah I'm at a "something is better than nothing" sort of stage on these.

File: cmd/warpforge/ware.go [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+					Name:  "force",
+					Usage: "Allow overwriting the given path. Any contents of an existing directory may be lost.",
+				},
+			},
+			Action: util.ChainCmdMiddleware(cmdWareUnpack,
+				util.CmdMiddlewareLogging,
+				util.CmdMiddlewareTracingConfig,
+				util.CmdMiddlewareTracingSpan,
+			),
+			ArgsUsage: "[WareID]",
```
[the remark this answers is not in the document]
Comment (TripleDogDare) [reply cmd/warpforge/ware.go:47]: Yeah that might be reasonable.

File: examples/500-cli/cli.md [hunk 3]
```diff
@@ -489,34 +495,54 @@ cat .warpforge/catalogs/default/github.com/githubtraining/training-manual/_mirro
...
-[testmark]:# (catalog/then-add/then-bundle/stdout)
+[testmark]:# (catalog/net/then-update/then-bundle/stdout)
 ```
 bundled "warpsys.org/busybox:v1.35.0:amd64-static"
 ```
+# Unpack
+
+[testmark]:# (catalog/net/then-update/then-unpack/sequence)
+```
+	warpforge ware unpack --path tmp tar:zM5K3W15SFfQZ5uJVdcEDgeHCoGhxLYLHKsMXvmUad4MUZ9raT2ropMsE66FqaeHDsaVWc7
```
[the remark this answers is not in the document]
Comment (TripleDogDare) [reply examples/500-cli/cli.md]: Probably just what my editor wanted to do when I made a new line. Can fix.

[the remark this answers is not in the document]
Comment (TripleDogDare) [reply examples/500-cli/cli.md]: Looks like I already did it. I guess that was the old commits.

File: cmd/warpforge/ware.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (TripleDogDare) [reply cmd/warpforge/ware.go:47]: [examples/500-cli/help.md](examples/500-cli/help.md)

**Review by ericevenchick — APPROVED** [review]
I've wanted this feature for a while :)

LGTM, works as expected for me. We probably do want to standardize a bit on help text and error text due to invalid args, but that's an _not here_ sort of problem.

**Pull request merged**